vendor/github.com/openshift/cluster-api/pkg/controller/machine/controller.go

```diff
@@ -145,6 +163,51 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, nil
 		}
 		klog.Infof("reconciling machine object %v triggers delete.", name)
+
+		// Drain node before deletion
+		// If a machine is not linked to a node, just delete the machine. Since a node
+		// can be unlinked from a machine when the node goes NotReady and is removed
+		// by cloud controller manager. In that case some machines would never get
+		// deleted without a manual intervention.
+		if _, exists := m.ObjectMeta.Annotations[ExcludeNodeDrainingAnnotation]; !exists && m.Status.NodeRef != nil {
+			if err := func() error {
+				kubeClient, err := kubernetes.NewForConfig(r.config)
```

Comment:
Maybe we should add the kubeClient to the reconciler as with do with oc client?

Reply:
I intentionally did no do that since this is only a temporary solution until controller-runtime implements `Patch`. Once that happens, the kube client initialization can be completely dropped.